test/test.js

```diff
@@ -470,6 +470,70 @@ exports['async next from consumer'] = function (test) {
     });
 };
 
+exports['generator throws error if next called after nil'] = function (test) {
+    test.expect(1);
+    var s = _(function(push, next) {
+        push(null, 1);
```

Comment:
You'll want some equivalent of the `nil_seen` check from below here too, so that the `next` doesn't cause a loop around and you end up throwing because of the `push(null, 1)` instead of the `next()`.